src/pdf_beamtime/src/pdf_beamtime_server.cpp

```diff
@@ -109,6 +124,12 @@ void PdfBeamtimeServer::handle_accepted(
   const std::shared_ptr<rclcpp_action::ServerGoalHandle<PickPlaceControlMsg>> goal_handle)
 {
   using namespace std::placeholders;
+
+  // Reset the fsm and move arm home before accepting new goals, except:
+  if (current_state_ != State::GRASP_SUCCESS || current_state_ == State::PLACE) {
+    reset_fsm();
+  }
+
```

Comment:
Doesn't this reset logic belong in the stop/abort cleanup?

Reply:
reset_fsm() is moved to the cleanup function.
https://github.com/maffettone/erobs/blob/d76de5584bf62ca3b3e65c1eac84eb61810aa109/src/pdf_beamtime/src/pdf_beamtime_server.cpp#L555-L576